src/app/(BottomTabNavigation)/Cases/QRCodeScanner/index.tsx

```diff
@@ -28,9 +42,28 @@ function QRCodeScannerScreen() {
   }, []);
 
   const handleBarCodeScanned = async (result: BarCodeScannerResult) => {
+    const caseId = result.data;
     if (!scanned) {
       setScanned(true);
-      setData(result.data);
+      const valid = await isValidCase(caseId);
+      if (!valid) {
+        // TODO: Display error toast message
+        setToast('Not a valid QRCODE!');
+        setScanned(false); // setTimeout here
+        return;
+      }
+      const duplicate = await containsDuplicateCase(caseId);
+      if (duplicate) {
+        setToast('DUPLICATES NOT ALLOWED!');
+        setScanned(false); // we should do setTimeout
```

Comment:
Hmm.. setting scanned to false here let's the user scan multiple times even if there is a toast component and we probably should not allow this. Same with invalid case. Probably need to have a bigger group discussion about this because it's hard to choose when to allow a user to rescan after scanning an invalid or duplicate code -- do we trust users to not spam it?